addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -101,6 +101,40 @@ function hasTurn(iceServers) {
     });
 }
 
+/**
+ * Checks whether hardware acceleration is available in the browser.
+ *
+ * This function attempts to create a WebGL rendering context, which requires GPU access.
+ * If WebGL is not supported, or a known software renderer (like SwiftShader or LLVMpipe)
+ * is detected, the function throws an error with a descriptive message.
+ */
+
+function checkHardwareAccelerationSupport() {
+    const canvas = document.createElement("canvas");
+    const gl =
+        canvas.getContext("webgl") ||
+        canvas.getContext("webgl2") ||
+        canvas.getContext("experimental-webgl");
+    if (!gl) {
+        const error = new Error(
+            "Your browser does not support WebGL. Hardware acceleration might be disabled in your system"
+        );
+        error.name = "Visual Effects Not Supported";
+        throw error;
+    }
+    const debugInfo = gl.getExtension("WEBGL_debug_renderer_info");
+    if (debugInfo) {
+        const renderer = gl.getParameter(debugInfo.UNMASKED_RENDERER_WEBGL);
+        if (/swiftshader|llvmpipe|software/i.test(renderer)) {
+            const error = new Error(
+                "Your system is using a software renderer. Visual effects require hardware acceleration to be enabled from your browser settings."
+            );
+            error.name = "Hardware acceleration is disabled";
+            throw error;
+        }
+    }
+}
```

Comment:
I would not throw error at all and permit usage of the feature anyway even without hardware acceleration available.
I would put a sticky alert on the camera card (with "X" to remove the alert) to tell that "Performance warning: WebGL Hardware Acceleration is off"

Browsers have hardware-acceleration enabled by default when the device has it, when people see this warning this can often come from device that is too old so they can't do much.

Reply:
There is no "official" way to check if the hardware acceleration is enabled and/or available. 
If I understand this correctly, this only checks that it's available, but not if it's enabled? I guess it's better than nothing, but it could lead to confusion about the fact that it can be disabled but still available.

Reply:
When available this is enabled by default. When disabled this was made by hand therefore the user is aware of making this change and there's no need to warn.

Reply:
We can't do much as you said we probably cannot detect whether the setting is on or off, only its availability.
People that are affected by the performance issue is mostly due to hardware not supporting it. Showing the warning only for this case is a significant help nonetheless.